crates/pathfinder/src/storage/schema/revision_0012.rs

```diff
@@ -0,0 +1,184 @@
+
+    tracing::info!("Re-created the starknet_events table and copied data");
+
+    transaction
+        .execute_batch(
+            r"
+            -- Event filters can specify ranges of blocks
+            CREATE INDEX starknet_events_block_number ON starknet_events(block_number);
+            
+            -- Event filter can specify a contract address
+            CREATE INDEX starknet_events_from_address ON starknet_events(from_address);",
+        )
+        .context("Recreating indexes for starknet_events")?;
+
+    tracing::info!("Re-created the indexes for starknet_events");
+
+    transaction
+        .execute_batch(r"
+            -- Drop FTS5 virtual table containing key lookup data
+            DROP TABLE starknet_events_keys;
+            
+            -- Re-create FTS5 virtual table as an external content table using `id` as the content rowid
+            CREATE VIRTUAL TABLE starknet_events_keys
+            USING fts5(
+                keys,
+                content='starknet_events',
+                content_rowid='id',
+                tokenize='ascii'
+            );
+            
+            -- Re-populate the full text index with keys
+            INSERT INTO starknet_events_keys (rowid, keys)
+                SELECT starknet_events.id, starknet_events.keys
+                FROM starknet_events;
+            
+            -- Re-create triggers updating the FTS5 table
+            CREATE TRIGGER starknet_events_ai
+            AFTER INSERT ON starknet_events
+            BEGIN
+                INSERT INTO starknet_events_keys(rowid, keys)
+                VALUES (
+                    new.id,
+                    new.keys
+                );
+            END;
+            
+            CREATE TRIGGER starknet_events_ad
+            AFTER DELETE ON starknet_events
+            BEGIN
+                INSERT INTO starknet_events_keys(starknet_events_keys, rowid, keys)
+                VALUES (
+                    'delete',
+                    old.id,
+                    old.keys
+                );
+            END;
+            
+            CREATE TRIGGER starknet_events_au
+            AFTER UPDATE ON starknet_events
+            BEGIN
+                INSERT INTO starknet_events_keys(starknet_events_keys, rowid, keys)
+                VALUES (
+                    'delete',
+                    old.id,
+                    old.keys
+                );
+                INSERT INTO starknet_events_keys(rowid, keys)
+                VALUES (
+                    new.id,
+                    new.keys
+                );
+            END;"
+        )
+        .context("Recreating the starknet_events_key FTS5 table and related triggers")?;
+
+    tracing::info!("Re-created the full-text index for starknet_events");
+
+    Ok(PostMigrationAction::None)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::storage::schema::{self, PostMigrationAction};
+    use rusqlite::Connection;
+
+    #[test]
+    fn empty() {
```

Comment:
For completeness sake - it's worth adding a simple stateful test:
- migrate to rev10, add an event, get the event,
- migrate to rev11, get the event again, assert_eq with previous result.

Reply:
Yeah, good idea.

I've added a stateful test.